### PR TITLE
Refactor GizmoMemberAccessor to use Member for metadata

### PR DESCRIPTION
**Pull request by Christopher-Chianelli**
- The constructor of the generated MemberAccessor
  get the declared field/method and uses it
  for the generic type and annotated element.
  It works in native mode currently (but
  classes are still registered for reflection
  since this doesn't have SolutionCloner yet).

- Additionally, generate a new class to initialize
  the Member Accessor map (meaning the generated
  MemberAccessor are no longer @ApplicationScoped)

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by Christopher-Chianelli** [comment]
Reading https://www.graalvm.org/reference-manual/native-image/Reflection/, the `getDeclaredField/getDeclaredMethod` calls should be okay (the generated bytecode look like MyClass.class.getDeclaredField("myField"))

File: optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoMemberDescriptor.java [hunk 1]
```diff
@@ -210,7 +230,41 @@ public String getName() {
...
+                case "F":
+                    return float.class.getName();
+                case "S":
+                    return short.class.getName();
+                case "D":
+                    return double.class.getName();
+                case "Z":
+                    return boolean.class.getName();
+                case "V":
+                    return void.class.getName();
```
Comment (ge0ffrey) [inline optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoMemberDescriptor.java:261]: Doesn't the JDK have anything to translate these for us?

File: optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/GizmoMemberAccessorEntityEnhancer.java [hunk 2]
```diff
@@ -247,6 +237,37 @@ private static String getTypeDescriptor(java.lang.reflect.Type type) throws Clas
...
+                    MethodDescriptor.ofMethod(Map.class, "put", Object.class, Object.class, Object.class),
+                    memberAccessorMap, generatedMemberAccessorResultHandle, memberAccessorInstance);
+        }
+        methodCreator.invokeStaticMethod(
+                MethodDescriptor.ofMethod(GizmoMemberAccessorFactory.class, "usePregeneratedMemberAccessorMap",
+                        void.class, Map.class),
+                memberAccessorMap);
+        methodCreator.returnValue(null);
+
+        classCreator.close();
```
Comment (ge0ffrey) [inline optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/GizmoMemberAccessorEntityEnhancer.java:267]: ClassCreator implements AutoCloseable, but this close() is't called when an exception occurs (which results in a memory leak).

Can we do ARM instead:

```
try (ClassCreator classCreator = ...) {

}
```

That will call close for us. Even if an exception occurs.

**Review by ge0ffrey — APPROVED** [review]
Over 2400 lines of code removed! Wonderful!!
See my AutoCloseable comment, but otherwise good to merge

File: optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoMemberDescriptor.java (hunk 1 above)
Comment (Christopher-Chianelli) [reply optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/common/accessor/gizmo/GizmoMemberDescriptor.java]: Gizmo didn't, but ASM did

**Pull request merged**